src/session/content/chat_action_bar.rs

```diff
@@ -560,3 +566,15 @@ fn message_bar_visibility_in_groups(
     )
     .upcast()
 }
+
+async fn parse_markdown(
+    formatted_text: types::FormattedText,
+    client_id: i32,
+) -> Option<types::FormattedText> {
+    let result = functions::parse_markdown(formatted_text, client_id).await;
+    if let Ok(text) = result {
+        let FormattedText::FormattedText(text) = text;
+        return Some(text);
+    }
+    None
```

Comment:
Try to use combinators. These are much more elegant than this 80s C-style if-else-return.

```suggestion
    functions::parse_markdown(formatted_text, client_id)
        .await
        .map(|text| {
            let FormattedText::FormattedText(text) = text;
            text
        })
        .ok()
```

Reply:
> Nitpick: Try to use combinators. These are much more elegant than this 80s if-else-return C-style.

Thanks for your advice